temoa/utilities/visualizer.py

```
@@ -0,0 +1,311 @@
+    """
+    dg = nx.MultiDiGraph()
+    node_styles_by_layer = {
+        1: {'size': 25, 'shape': 'ellipse'},
+        2: {'size': 15, 'shape': 'dot'},
+        3: {'size': 20, 'shape': 'box'},
+    }
+    default_node_style = {'size': 15, 'shape': 'dot'}
+
+    # 1. Add all nodes first to ensure they exist before adding edges
+    all_nodes_in_connections = {c[0] for c in connections} | {c[2] for c in connections}
+    for node_name in sorted(list(all_nodes_in_connections)):  # Sort for consistency
```

Ruff (0.14.1)

Unnecessary `list()` call within `sorted()`

Remove the inner `list()` call

(C414)